openstack/loadbalancer/v2/flavorprofiles/requests.go

```diff
@@ -0,0 +1,111 @@
+package flavorprofiles
+
+import (
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/pagination"
+)
+
+// LIST
+
+type ListOptsBuilder interface {
+	ToFlavorProfileListQuery() (string, error)
+}
+
+type ListOpts struct {
+	ID   string `q:"id"`
+	Name string `q:"name"`
+}
+
+func (opts ListOpts) ToFlavorProfileListQuery() (string, error) {
+	q, err := gophercloud.BuildQueryString(opts)
+	return q.String(), err
+}
+
+func List(c *gophercloud.ServiceClient, opts ListOptsBuilder) pagination.Pager {
+	url := rootURL(c)
+	if opts != nil {
+		query, err := opts.ToFlavorProfileListQuery()
+		if err != nil {
+			return pagination.Pager{Err: err}
+		}
+		url += query
+	}
+	return pagination.NewPager(c, url, func(r pagination.PageResult) pagination.Page {
+		return FlavorProfilePage{pagination.LinkedPageBase{PageResult: r}}
+	})
+}
+
+// CREATE
+
+type CreateOptsBuilder interface {
+	ToFlavorProfileCreateMap() (map[string]interface{}, error)
+}
+
+type CreateOpts struct {
+	Name         string `json:"name,omitempty"`
```

Comment:
`name` is required too, isn't it?

Also, please make sure to [document the fields](https://github.com/gophercloud/gophercloud/blob/master/docs/contributor-tutorial/step-05-pull-requests.md#documentation).

Reply:
Done